### PR TITLE
#162633478, #162633477  Implement Bookmarking Functionality

### PR DESCRIPTION
**Pull request by NabunyaLilian**
**What does this PR do?**

Implements Bookmark functionality feature in articles.

**Description of Task to be completed?**

- Create components for bookmarking and listing bookmarks.
- Create actions and reducers for bookmarks.
- Create tests for the feature.

**How should this be manually tested?**
- Run `npm install` to install dependencies.
- Run `npm test` to test the functionality.
- Run `npm start` to run the application.
- Check your redux dev tools to confirm that the state in your app appears in the dev tools.

**What are the relevant pivotal tracker stories?**

[#162633478](https://www.pivotaltracker.com/story/show/162633478)
[#162633477](https://www.pivotaltracker.com/story/show/162633477)

### CODE REVIEW

File: src/components/Bookmarks/actions/bookmark_actions.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+export const deleteBookmarkError = error => (
+    {
+        type: BookmarkActionTypes.DELETE_BOOKMARK_ERROR,
+        payload: error,
+    }
+);
+
+export const bookmarkAction = slug => ( dispatch ) => {
+    // const url = `${ process.env.HOST }/articles/${slug}/bookmark`
```
Comment (mubstimor) [inline src/components/Bookmarks/actions/bookmark_actions.js]: please remove commented out code that's unused.

File: src/components/Bookmarks/bookmarkArticle.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+    handleOnClick = () => {
+        const { articleSlug } = this.props;
+        const {bookmarkedArticleId} = this.state;
+        if(bookmarkedArticleId===0){
+            this.props.bookmarkAction( articleSlug);
+        }else{
+            this.props.unBookmark(articleSlug,bookmarkedArticleId)
+        }
+
```
Comment (mubstimor) [inline src/components/Bookmarks/bookmarkArticle.js]: blank line padding not necessary

File: src/components/Bookmarks/bookmarkArticle.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        if (nextProps.bookmarks.results){
+            const results = nextProps.bookmarks.results;
+            const foundValue = results.filter(bookmark=>bookmark.article.slug===articleSlug);
+            if (foundValue.length>0){
+                this.setState({ icon_state:'fas fa-bookmark text-primary', bookmarkedArticleId:foundValue[0].id })
+            }else{
+                this.setState({ icon_state:'fas fa-bookmark text-secondary' })
+            }
+        }
+
```
Comment (mubstimor) [inline src/components/Bookmarks/bookmarkArticle.js]: same blank line issue

**Comment by mubstimor** [comment]
please squash the commits into one

**Comment by NabunyaLilian** [comment]
Thanks for the feedback but as the label says my PR is still WIP, I will work on it.

**Comment by ruganda** [comment]
@NabunyaLilian , Please resolve the conflicts

**Pull request merged**